domain/secret/state/state.go

```diff
@@ -2459,3 +2466,165 @@ FROM secret_revision sr`, selectStmt)
 	})
 	return errors.Trace(err)
 }
+
+type (
+	revisions     []int
+	revisionUUIDs []string
+)
+
+// DeleteSecret deletes the specified secret revisions.
+// If revisions is nil or the last remaining revisions are removed.
+func (st State) DeleteSecret(ctx context.Context, uri *coresecrets.URI, revs []int) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	// First delete the specified revisions.
+
+	selectRevisionParams := []any{secretID{
+		ID: uri.ID,
+	}}
+
+	var revFilter string
+	if len(revs) > 0 {
+		revFilter = "\nAND revision IN ($revisions[:])"
+		selectRevisionParams = append(selectRevisionParams, revisions(revs))
+	}
+
+	selectRevsToDelete := fmt.Sprintf(`
+SELECT uuid AS &revisionUUID.uuid
+FROM   secret_revision
+WHERE  secret_id = $secretID.id%s
+`, revFilter)
+	selectRevisionStmt, err := st.Prepare(selectRevsToDelete, secretID{}, revisionUUID{}, revisions{})
```

Comment:
`revisions{}` is only required if len(rev) > 0.
Just use the `append([]any{revisionUUID{}}, selectRevisionParams...)`.

Reply:
Still works as is :-) I had the append originally but thought it looked cleaner without. But changed.